### PR TITLE
Added #863: Explore and Feed

### LINKED ISSUE #863: Explore and Feed views
We're going to have a new view on the dashboard.
The view would have multiple tabs to show different data.

The first one would be Explore tab which would show all the published entries on the platform that user can see.

The second tab is called "Feed" where user will be seeing published entries from the websites that is subscribed to.

In both views user can read, like, comment and share in the same place and doesn't need to open any other link.

The endpoints: 

* Explore: /api/v1/sushial/explore/
* Feed (Subscribed posts): /api/v1/sushial/subscribed-entries/

### PR DESCRIPTION
**Pull request by ArsalanSavand**
Fixes #863 

- Main view
image

- Blog creation when user has no blog
image

- Reader when user has no blog
image

### CODE REVIEW

File: src/app/app_routes.js [hunk 1]
```diff
@@ -187,6 +187,13 @@ angular.module("gonevisDash").config(function ($stateProvider, $urlRouterProvide
       auth: true,
       title: "Posts"
     })
+    .state("dash.reader", {
+      url: "/reader",
+      controller: "ReaderController",
+      template: require("./reader/reader_view.html"),
+      auth: true,
+      title: "Reader"
```
Comment (Alir3z4) [inline src/app/app_routes.js]: Does this requires translation as well ?

**Comment by Alir3z4** [comment]
OK, I'm going to try it now and see how it is.

**Comment by Alir3z4** [comment]
OK, no worries, I'm writing my full review and further requests on this
pull request. You gonna love it :D

File: src/app/app_routes.js (hunk 1 above)
Comment (Alir3z4) [reply src/app/app_routes.js]: OK, no worries, I'm writing my full review and further requests on this
pull request. You gonna love it :D

**Comment by Alir3z4** [comment]
Excellent work on the header, nice design, especially the animation:

image

(_Please remove the emoji and icons, it doesn't fit into the design._)

It looks really good, even though it requires some enhancement to be polished but it's great for the start.

Now, stuff that need to be added:

* [ ] When clicking on each card/post, the post detail should be on dashboard, we don't want the user to send off on the main blog.
  * [ ] leave n small link/button saying "Visit Site" on top right of the detail page that sends user to the main address of the post.
  * [ ] If user click on the post/entry card, post should be displayed in the dashboard.
  * [ ] If user tries to open the link in the new tab, like `middle-click` then user should be sent to the actual site (absoulte_uri)
* [ ] Make it infinite load, no button and don't let user reach the end of page to load the next page, load it long before they reach end of result.
* [ ] Add a List/View (_Optional, can be done in separate pull request._)
* [ ] Outline what minimal data you would to have when when getting the list response, because the data you're receiving are detailed and include lots of stuff that you don't use right now on the reader, most of them are useful when showing the detail page.
* [ ] Move Reader to its own tab/view, completely separated from the Blog management mode.
* [ ] Let the excerpt/lead content should not have any html rendering, this should be handled by the backend, make an issue in the backend to provide that specific data.
* [ ] Remove the glowing effect from the post owner avatar, it makes me thing it's something new or special to checkout and it can be become annoying quickly.

### Reader & Site mode Layout

image

The READER should not be part of the blog navigation. Any link in the navigation is a property under the current blog you have selected to manager.

In this case, we need to make a better UX to separate this 2 modes, when having Reader, GoNevis Dashboard will need to have a separate mode, which we call Reader and the current one wil be Manage or Blog model.

We can distinguish these 2 mode by a link in the top header of the dashboard, something like:

image

Because in this case user knows Reader is not part of managing the blog they own or part of an org, but a separate view and complete set of features used for their exploring and finding new content.

Another strong point for having Reader separated is because eventually it will be a complete module that will have many features and inner views and tabs itself, for instance things that will be included would be:

* Topics/Tags they follow; in that case user will be able to pick select a tag and only see stuff related to that tag.
* Search view
* etc...
* A sidebar to collect all the view

In that case having Reader part of Blog navigation will not be benefiting us in any way.

**Comment by ArsalanSavand** [comment]
@Alir3z4 about your first request, can we do that in another pull request?

**Comment by Alir3z4** [comment]
> @Alir3z4 about your first request, can we do that in another pull request?

Which one especially ? the post detail page ?

You can merge this and start working on other ones, each in a separate pull request.

However, don't merge into master.
Make a new branch called "reader" and merge all the pull request into that.
Once the whole feature is ready, merge the "reader" into master.

Makes sense ?

**Comment by Alir3z4** [comment]
@ArsalanSavand I just made a new branch called "reader", and change the base branch of this pull request to is.

If you merge this, it will go into that branch and won't touch the master.
In this case, feel free to break this feature into smaller pull requests and merge one by one into that branch so you can work easier and the pull request doesn't get big and annoying to handle.

In this case, go ahead and make a task for each feature request I made in the comments and work on each individually and make pull request against "reader" branch.

Makes sense ?

**Comment by ArsalanSavand** [comment]
Ok sounds good.

**Comment by Alir3z4** [comment]
@ArsalanSavand what's the update on this feature ?
I can't wait to see it live

**Comment by Alir3z4** [comment]
**Sidebar & Navbar**

Here's some suggestion to make the changes on navbar and sidebar feel more integrated.

* [ ] Let the sidebar has a darker color of the main container. Darker enough to be easily distinguishable from the main container.
* [ ] Navbar can be either completely dark or blue or any color, I think light blue anything works, as long it's not hurting eyes, the color should be smooth.
* [ ] Move the **Blog selector** to the Navbar completely, I think it would make sense to replace the Blog Selector with "Dashboard" label in the navbar. 
  * [ ] If there's only 1 blog, don't show the drop-down, once clicked on the blog name, redirect to the its dashboard.
 * [ ] If there's more than 1 blog, show a tiny drop-down icon and let the user select a different blog to get into the blog's dashboard.
* [ ] Move **Feedback** from Account dropdown of the navbar and put it next to the User avatar.
* [ ] Any general button or info should go on the navbar.
  * [ ] Blog Selector
  * [ ] Reader
  * [ ] User Account...
  * [ ] Feedback

File: src/app/account/user/user_controller.js [hunk 2]
```diff
@@ -45,7 +45,9 @@ function UserController($scope, $rootScope, $stateParams,
         } else {
           $scope.user[key] = data[key];
         }
-        $scope.user = AuthService.setAuthenticatedUser($scope.user.get, true);
+
+        $scope.user = new Account(data);
+        $scope.user = AuthService.setAuthenticatedUser(data, true);
```
Comment (AmirSavand) [inline src/app/account/user/user_controller.js]: Why?

**Comment by ArsalanSavand** [comment]
@Alir3z4 In the blog selection dropdown, there is a link for creating a new blog. So we can't use the method you are talking about.

**Comment by Alir3z4** [comment]
> @Alir3z4 In the blog selection dropdown, there is a link for creating a new blog. So we can't use the method you are talking about.

I don't follow, could you elaborate more please ?

File: src/app/account/user/user_controller.js (hunk 2 above)
Comment (Alir3z4) [reply src/app/account/user/user_controller.js]: Yes, exactly, why ?
Is this intentional or pure trolling ?

**Comment by ArsalanSavand** [comment]
@Alir3z4 you said: 
"If there's only 1 blog, don't show the drop-down, once clicked on the blog name, redirect to the its dashboard."

We have to show the drop-down icon so that user knows he can have multiple sites and beside that, "Create a new blog" is also in the dropdown

File: src/app/account/user/user_controller.js (hunk 2 above)
Comment (ArsalanSavand) [reply src/app/account/user/user_controller.js]: This code will help update sites on navbar.

**Pull request merged**